### PR TITLE
Feat/message inputonly outputonly

### PR DESCRIPTION
**Pull request by ppaanngggg**
https://stackoverflow.com/questions/33629750/swagger-schema-properties-ignored-when-using-ref-why

As this answer, add `allOf` to wrap `$ref` can support siblings for message field.

### CODE REVIEW

**Comment by morphar** [comment]
Thank @ppaanngggg 👍 
I am thinking, that maybe this should actually be default behaviour.
What are your thoughts @timburks?

**Comment by ppaanngggg** [comment]
> default behaviour.

I don't think...  It will be a big break change

**Comment by timburks** [comment]
@ppaanngggg could you explain more about why this is a breaking change? Like @morphar, I'd like to not have another option because we'll wind up with a lot of cases to test and maintain.

**Comment by ppaanngggg** [comment]
@timburks 
There are some misunderstandings.
1. no need to migrate to OpenAPI 3.1
2. this is a way to support $ref siblings in OpenAPI 3.0.x
3. this is a valid OpenAPI, https://editor.swagger.io/ can render it, I copy openapi_allof_wrap.yaml into this editor

**Comment by ppaanngggg** [comment]
@timburks @morphar review this mr plz.
I have use this for a while, I think it works well.

**Comment by timburks** [comment]
@ppaanngggg Thanks, I've done some additional checking: I ran your test example through a schema validator and reviewed the 3.0.3 spec). Based on that, I accept that this is valid OpenAPI. 

Back to @morphar's comment, I would like to find a way to include this without adding an option. Could we automatically add the allOf wrapper only when needed? (see my comment inline)

File: cmd/protoc-gen-openapi/generator/generator.go [hunk 1]
```diff
@@ -786,6 +787,15 @@ func (g *OpenAPIv3Generator) addSchemasForMessagesToDocumentV3(d *v3.Document, m
 				continue
 			}
 
+			if *g.conf.AllOfWrap {
```
[the remark this answers is not in the document]
Comment (ppaanngggg) [reply cmd/protoc-gen-openapi/generator/generator.go:790]: In fact, this allof wrapper can support any siblings as `schema` struct. But the most useful properties is `inputOnly` and `outputOnly`

**Comment by ppaanngggg** [comment]
> @ppaanngggg Thanks, I've done some additional checking: I ran your test example through a schema validator and reviewed the 3.0.3 spec). Based on that, I accept that this is valid OpenAPI.
> 
> Back to @morphar's comment, I would like to find a way to include this without adding an option. Could we automatically add the allOf wrapper only when needed? (see my comment inline)

Great idea

**Comment by ppaanngggg** [comment]
@timburks Fixed! Please check again.

**Review by timburks — APPROVED** [review]

**Comment by timburks** [comment]
LGTM - @morphar do you have any remaining observations before we merge this?

**Comment by morphar** [comment]
LGTM too 👍 

Nice with the last change to make it automatic.
Thank you @ppaanngggg :)

**Pull request merged**